fhir-persistence-blob/src/main/java/com/ibm/fhir/persistence/blob/FHIRPayloadPersistenceBlobImpl.java

```diff
@@ -71,25 +75,55 @@ public PayloadPersistenceResponse storePayload(String resourceType, int resource
             int version, String resourcePayloadKey, List<String> elements) throws FHIRPersistenceException {
 
         logger.fine(() -> "readResource " + rowResourceTypeName + "[" + resourceTypeId + "]/" + logicalId + "/_history/" + version);
-        final BlobPropertyGroupAdapter config = getConfigAdapter();
         BlobReadPayload cmd = new BlobReadPayload(resourceTypeId, logicalId, version, resourcePayloadKey, elements, !PAYLOAD_COMPRESSED);
-        return cmd.run(resourceType, getBlobContainerClient());
+        try {
+            // synchronous read
+            return cmd.run(resourceType, getBlobManagedContainer()).get();
```

Comment:
is this synchronous readResource flavor still being used somewhere?

Reply:
Yes, it's used from convertResourceDTO which gets called when we need the resource right away. For example, UPDATE requires that we read the current value of the resource so that we can emit the necessary before-update events. Same for PATCH etc. This makes these interactions take a little longer (because now we have to read from the database, then read the blob) but overall it shouldn't impact scalability. There are some tricks that we could do in the future. We could potentially read-ahead the resource payload before we hit the RDBMS, then check the resourcePayloadKey matches. But it actually requires two reads to the blob service (the first one being a listBlobs call) so I don't think we'd gain much in the end (the RDBMS read itself is very quick anyway).